--- swt-gtk-3.4.2.orig/make_linux.mak
+++ swt-gtk-3.4.2/make_linux.mak
@@ -50,6 +50,7 @@
 
 CDE_LIBS = -L$(CDE_HOME)/lib -R$(CDE_HOME)/lib -lXt -lX11 -lDtSvc
 
+AWT_LIB_PATH=/usr/lib/jvm/default-java/lib
 AWT_LIBS = -L$(AWT_LIB_PATH) -ljawt -shared
 
 ATKCFLAGS = `pkg-config --cflags atk gtk+-2.0`
@@ -64,6 +65,7 @@
 # Uncomment for Native Stats tool
 #NATIVE_STATS = -DNATIVE_STATS
 
+JAVA_HOME=/usr/lib/jvm/default-java
 MOZILLACFLAGS = -O \
 	-DSWT_VERSION=$(SWT_VERSION) \
 	$(NATIVE_STATS) \
@@ -76,8 +78,10 @@
 	-I. \
 	-I$(JAVA_HOME)/include \
 	-I$(JAVA_HOME)/include/linux \
+	`pkg-config --cflags libxul-embedding` \
 	${SWT_PTR_CFLAGS}
-MOZILLALIBS = -shared -Wl,--version-script=mozilla_exports -Bsymbolic
+MOZILLALIBS = -shared -Wl,--version-script=mozilla_exports -Bsymbolic \
+	`pkg-config --libs libxul-embedding`
 MOZILLAEXCLUDES = -DNO_XPCOMGlueShutdown -DNO_XPCOMGlueStartup
 	
 SWT_OBJECTS = swt.o c.o c_stats.o callback.o
@@ -92,7 +96,7 @@
 XPCOMINIT_OBJECTS = swt.o xpcominit.o xpcominit_structs.o xpcominit_stats.o
 GLX_OBJECTS = swt.o glx.o glx_structs.o glx_stats.o
 
-CFLAGS = -O -Wall \
+override CFLAGS += \
 		-DSWT_VERSION=$(SWT_VERSION) \
 		$(NATIVE_STATS) \
 		-DLINUX -DGTK \
@@ -102,6 +106,7 @@
 		${SWT_PTR_CFLAGS}
 LIBS = -shared -fPIC
 
+NO_STRIP=1
 ifndef NO_STRIP
 	AWT_LIBS := $(AWT_LIBS) -s
 	MOZILLALIBS := $(MOZILLALIBS) -s
