diff --git a/org/eclipse/swt/accessibility/AccessibleObject.java b/org/eclipse/swt/accessibility/AccessibleObject.java
index 00b05d5..1111cbd 100644
--- a/org/eclipse/swt/accessibility/AccessibleObject.java
+++ b/org/eclipse/swt/accessibility/AccessibleObject.java
@@ -1176,7 +1176,10 @@ class AccessibleObject {
 	}
 
 	static AccessibleObject getAccessibleObject (int /*long*/ atkObject) {
-		return (AccessibleObject)AccessibleObjects.get (new LONG (atkObject));
+		AccessibleObject object = (AccessibleObject)AccessibleObjects.get (new LONG (atkObject));
+		if (object == null) return null;
+		if (object.accessible == null) return null;
+		return object;
 	}
 	
 	AccessibleObject getChildByHandle (int /*long*/ handle) {
