diff -Nurp swt-gtk-3.6.2.orig//make_linux.mak swt-gtk-3.6.2/make_linux.mak
--- swt-gtk-3.6.2.orig//make_linux.mak	2011-03-21 11:12:43.205883562 +0000
+++ swt-gtk-3.6.2/make_linux.mak	2011-03-21 11:23:02.149147482 +0000
@@ -95,7 +95,7 @@ MOZILLAEXCLUDES = -DNO__1XPCOMGlueShutdo
 XULRUNNEREXCLUDES = -DNO__1NS_1InitXPCOM2
 
 WEBKITCFLAGS = `pkg-config --cflags gtk+-2.0` -I/usr/include/webkit-1.0 -I/usr/include/libsoup-2.4
-WEBKITLIBS = -lwebkit-1.0
+WEBKITLIBS = `pkg-config --libs webkit-1.0`
 
 SWT_OBJECTS = swt.o c.o c_stats.o callback.o
 CDE_OBJECTS = swt.o cde.o cde_structs.o cde_stats.o
